### ISSUE TITLE
Add functionality to copy logs, screenshots, and other artifacts to LUCI recipe location in engine repo

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#74344]
> Recipe will run the test in a deferred context which will create an env variable called FLUTTER_LOGS_DIR, the test runner can place any logs|artifacts to help debugging there and they will be automatically uploaded to GCS.

https://github.com/flutter/flutter/issues/74332#issuecomment-763865412

See also https://github.com/flutter/flutter/issues/43124

**Comment by jmagman** [comment]
@godofredoc Let me know when you have the recipe and modules updated with `FLUTTER_LOGS_DIR`, I have a PR to test it out at https://github.com/flutter/flutter/pull/74378

**Comment by jmagman** [comment]
@godofredoc This will be helpful in the engine as well as the framework. 😄

**Comment by jmagman** [comment]
@godofredoc this is done in the engine as well, right? Can we close?

**Comment by godofredoc** [comment]
Let me check, I don't remember if it was added to the engine recipe.

**Comment by godofredoc** [comment]
The logic to collect logs has not been added to the engine yet.

**Comment by godofredoc** [comment]
LUCI playbook has been updated with the instructions to clean mac caches. The caches of `build866-m9` has already been cleaned.

Please reopen if you find another builder failing for the same reason.

**Comment by jmagman** [comment]
@godofredoc is there any way we can prioritize getting this done in the engine repo?  It's really hard to debug failures like https://github.com/flutter/engine/pull/26786#issuecomment-862724209 without the uploaded test results.

**Comment by godofredoc** [comment]
I was actually blocking this on separating the builds/tests in the recipes. The difficulty is that we need to run the step to upload logs in a context to ensure logs are uploaded even if a test fails. Unfortunately the engine recipe has tests everywhere forcing the implementation in several places.

If this is a blocker I'll give a try to implement the logic in the current version of the recipes.

**Comment by jmagman** [comment]
No, it's not a blocker.  Thanks for the explanation, looking forward to the shiny new separated recipe world!

**Comment by jmagman** [comment]
We tested running the engine iOS tests in parallel, and unfortunately in that mode the tests don't report errors or stack traces, and the only way to get that info would be from an uploaded `xcresult` bundle.

See https://github.com/flutter/engine/pull/34020#issuecomment-1156999600.

That means this is now blocking https://github.com/flutter/flutter/issues/106050 for the engine tests part.

**Comment by jmagman** [comment]
This is also needed for plugin and packages since they are migrating from cirrus to luci.  Would be really helpful to have screenshots of native failures like https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8795950591422548017/+/u/Run_plugin_tests/native_test/stdout
cc @stuartmorgan

**Comment by jmagman** [comment]
@godofredoc should I split off this work between engine and plugins?  I think the engine work is blocked on the build/tests in the engine recipes, but I don't think there's any such work planned for plugins.  It's so hard to figure out why tests failed without being to upload test results with screenshots, crash logs, etc.

**Comment by godofredoc** [comment]
Checking if we already implemented this in engine v2 recipes.

**Comment by godofredoc** [comment]
Implementation for engine v2:

https://flutter-review.googlesource.com/c/recipes/+/39100

**Comment by jmagman** [comment]
> This is also needed for plugin and packages since they are migrating from cirrus to luci.

Merging of the plugins and packages repo is imminent, so we would only need this for flutter/packages recipes.

**Comment by jmagman** [comment]
Breaking the packages request off to https://github.com/flutter/flutter/issues/121535 since it needs to be implemented in separate recipes.

**Comment by godofredoc** [comment]
Closed with https://flutter-review.googlesource.com/c/recipes/+/39703
